Bosses.py

```diff
@@ -26,18 +27,19 @@ def ArmosKnightsDefeatRule(state, player):
         state.has('Red Boomerang', player))
 
 def LanmolasDefeatRule(state, player):
-    # TODO: Allow the canes here?
     return (
         state.has_blunt_weapon(player) or
         state.has('Fire Rod', player) or
         state.has('Ice Rod', player) or
+        state.has('Cane of Somaria', player) or
+        state.has('Cane of Byrna', player) or
```

Comment:
Do the canes here assume some amount of bombs?

Reply:
Byrna does require bombs (but less than 10 when I tested it) - we can make that require magic extension if you don't want that.  Somaria you don't even need full magic.

This is also the website logic.

Reply:
Byrna needs about 4 bombs (8 when I completely screwed up my magic usage).  12 bombs is actually enough for the fight without any other weapons assuming no double hits.

Reply:
Guess I haven't done bomb lanmo in forever, that doesn't sound so bad.